src/csharp/Sessionless/Impl/MessageSignature.cs

```diff
@@ -50,6 +50,11 @@ public record MessageSignatureHex : IMessageSignature {
         RHex = rHex; SHex = sHex;
     }
 
+    public MessageSignatureHex(string signature) {
```

Comment:
# Duplicate constructor
 `MessageSignatureHex(string)` is identical to `MessageSignatureHex(string,int)` where int parameter is optional (with default value)
```csharp
    public MessageSignatureHex(string signature) {...}
    public MessageSignatureHex(string rsHex, int partSize = 32)
                : this(rsHex[..partSize], rsHex[partSize..]) {
        //...
    }
```

# Solution: 
use the existing constructor (`MessageSignatureHex(string rsHex,int partSize=32)`) and optionally give it the length of the key parts 

Note: according to your code, a part is 64 long then partSize default value shall become 64